src/map/vending.cpp

```diff
@@ -200,8 +216,7 @@ void vending_purchasereq(struct map_session_data* sd, int aid, int uid, const ui
 
 	pc_payzeny(sd, (int)z, LOG_TYPE_VENDING, vsd);
 	achievement_update_objective(sd, AG_SPEND_ZENY, 1, (int)z);
-	if( battle_config.vending_tax )
-		z -= z * (battle_config.vending_tax/10000.);
+	vending_calc_tax(sd, z);
```

Comment:
For clarification it's better to use `z = vending_calc_tax(sd, z);` instead of just returning void.

Reply:
if he do that, it's better to remove zeny pass as reference to avoid any possible confusion,
param2= zenyin
return = zenytaxed.